dr_ensenso/src/opencv.cpp

```diff
@@ -40,52 +18,50 @@ cv::Mat toCvMat(NxLibItem const & item, std::string const & what) {
 	return result;
 }
 
-cv::Mat toCameraMatrix(NxLibItem const & item, std::string const & camera, std::string const & what) {
-	cv::Mat result;
-	try {
-		result = cameraMatrixImpl(item[itmCalibration][itmMonocular][camera == "Left" ? itmLeft : itmRight], what);
-	} catch (NxError const & e) {
-		throw e;
+cv::Mat cameraMatrixImpl(NxLibItem const & item, std::string const & what) {
+	int error = 0;
+	cv::Mat result = cv::Mat::zeros(3, 3, CV_64F);
+	for (std::size_t i=0; i<3; i++) {
+		for (std::size_t j=0; j<3; j++) {
+			result.at<double>(i,j) = item[itmCamera][j][i].asDouble(&error);
+			if (error) throw NxError(item, error, what);
+		}
```

Comment:
Can't we just get rid of the difference between `Impl` and the wrappers and expose the `Impl` versions directly (without `Impl` in the name)?

Selecting the right `NxLibItem` would then be up to the caller, which is fine with me.

Note that if you do want to compare a `std::string`, then `string == "value"` is much more idiomatic in C++ than `string.compare("value") == 0`. The latter is more C-ish.